idaes/power_generation/control/pid_controller.py

```diff
@@ -0,0 +1,208 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+PID controller block, Revised to use pyomo.DAE
+to calculate integral and differential parts
+"""
+
+__author__ = "John Eslick and Jinliang Ma"
+
+
+from pyomo.environ import Var, Param, Constraint, Reference,\
+    exp, log
+from pyomo.dae import DerivativeVar
+
+from idaes.core import UnitModelBlockData, declare_process_block_class
+from pyomo.common.config import ConfigValue, In
+from idaes.core.util.exceptions import ConfigurationError
+
+
+@declare_process_block_class("PIDController", doc="""This is a PID controller
+                             block. Based on UnitModelBlockData to configure
+                             the dynamic flag based on parent""")
+class PIDControllerData(UnitModelBlockData):
+    CONFIG = UnitModelBlockData.CONFIG()
+    CONFIG.declare("pv", ConfigValue(
+        default=None,
+        description="Process variable to be controlled",
+        doc="A Pyomo Var, Expression, or Reference for the measured"
+            " process variable. Should be indexed by time."))
+    CONFIG.declare("mv", ConfigValue(
+        default=None,
+        description="Manipulated process variable",
+        doc="A Pyomo Var, Expression, or Reference for the controlled"
+            " process variable. Should be indexed by time."))
+    CONFIG.declare("bounded_output", ConfigValue(
+        default=False,
+        description="Flag to bound manipulated variable",
+        doc="""Indicating if the output for the manipulated variable is bounded
+Default: False.
+If True, user need to set the lower and upper bound parameters"""))
+    CONFIG.declare("type", ConfigValue(
+        default="PI",
+        domain=In(['P', 'PI', 'PD', 'PID']),
+        description="Control type",
+        doc="""Controller type options including
+- P: Proportional only
+- PI: Proportional and integral only
+- PD: Proportional and derivative only
+- PID: Proportional, integral and derivative
+Default is PI"""))
+
+    def build(self):
+        """
+        Build the PID block
+        """
+        super().build()
+
+        # Do nothing if steady-state
```

Comment:
Very minor, but would it have been clearer to do:

```
if not self.config.dynamic:
    return
```

Also, should we log a message to say we did mothing (info_high or debug level)?